scratch/ingest.py

```diff
@@ -104,65 +194,59 @@ def main() -> None:
 
 async def async_main(args: argparse.Namespace) -> None:
     logger = logging.getLogger(__name__)
-    host, port = await get_product_controller_endpoint(args.mc_address, args.product_name)
+    if args.corr2:
+        host, port = await get_subordinate_endpoint(args.mc_address, args.product_name)
+    else:
+        host, port = await get_product_controller_endpoint(args.mc_address, args.product_name)
     client = await aiokatcp.Client.connect(host, port)
 
     async with client:
+
+        async def sensor_val(name):
+            if args.corr2:
+                name = CORR2_SENSOR_REMAP.get(name, name.replace(".", "-"))
+            return await get_sensor_val(client, name)
```

Comment:
An alternative solution would be to make `corr2` another argument to `get_sensor_val` (and move the functionality for corr2 in there), and then instead of writing a wrapper function here, just do something like:
```python
sensor_val = functools.partial(get_sensor_val, client, args.corr2)
```
I'm not sure if you think that's any cleaner.

Reply:
I did think of adding a `corr2` argument, but that seemed clunky to me because then every line would need `args.corr2` in it which would be more typing.

It doesn't bother me enough to hold up the merge. It clearly does the job. If I think of a better solution I'll propose it. In the meantime, I think you can go ahead and merge.